gravitee-apim-console-webui/src/portal/customization/banner/portal-banner.component.ts

```diff
@@ -137,4 +154,47 @@ export class PortalBannerComponent implements OnInit {
       )
       .subscribe(() => this.ngOnInit());
   }
+
+  private initializeButtonForm(button: BannerButton | undefined): FormGroup<ButtonForm> {
+    return new FormGroup<ButtonForm>({
+      enabled: new FormControl<boolean>(button?.enabled ?? false, [Validators.required]),
+      label: new FormControl<string>(button?.label ?? ''),
+      visibility: new FormControl<string>(button?.visibility ?? 'PUBLIC'),
+      target: new FormControl<string>(button?.target ?? ''),
+    });
+  }
+
+  private handleButtonFormEnabledChanges(buttonForm: FormGroup<ButtonForm>, startWithEnabledValue: boolean): void {
+    buttonForm.controls.enabled.valueChanges
+      .pipe(
+        startWith(startWithEnabledValue),
+        tap((_) => {
+          if (buttonForm.controls.enabled.value) {
+            buttonForm.controls.target.addValidators(Validators.required);
+            buttonForm.controls.label.addValidators(Validators.required);
+            buttonForm.controls.visibility.addValidators(Validators.required);
+          } else {
+            buttonForm.controls.target.clearValidators();
+            buttonForm.controls.label.clearValidators();
+            buttonForm.controls.visibility.clearValidators();
+          }
+          buttonForm.controls.target.updateValueAndValidity();
+          buttonForm.controls.label.updateValueAndValidity();
+          buttonForm.controls.visibility.updateValueAndValidity();
+        }),
+        takeUntilDestroyed(this.destroyRef),
+      )
+      .subscribe();
+  }
+
+  private prepareButtonForUpdate(buttonForm: FormGroup<ButtonForm>): BannerButton {
+    const rawValue = buttonForm.getRawValue();
+    return {
+      enabled: rawValue.enabled,
+      target: rawValue.target,
+      visibility: rawValue.visibility === 'PRIVATE' ? 'PRIVATE' : 'PUBLIC',
```

Comment:
Good that you ser the same default value as `PUBLIC` in both cases. Here and 
here: `visibility: new FormControl<string>(button?.visibility ?? 'PUBLIC'),`